provider/bidengine/order.go

```diff
@@ -404,7 +404,13 @@ func (o *order) shouldBid(group *dtypes.Group) (bool, error) {
 
 	// does provider have required attributes?
 	if !group.GroupSpec.MatchAttributes(o.session.Provider().Attributes) {
-		o.log.Debug("unable to fulfill: incompatible attributes")
+		o.log.Debug("unable to fulfill: incompatible provider attributes")
+		return false, nil
+	}
+
+	// does order have required attributes?
+	if !o.cfg.Attributes.SubsetOf(group.GroupSpec.Requirements.Attributes) {
```

Comment:
Does `MatchAttributes` just call `SubsetOf`?  Let's use the exact same matching method if not.

Reply:
I'm unsure why would need this additional check, if there is a bug its probably in `MatchAttributes` and should be fixed there

Reply:
Both `MatchAttributes` and `SubsetOf` internally just call `AttributesSubsetOf`. So, it seems fine to keep it as it is.